src/sentry_core.c

```diff
@@ -134,22 +157,22 @@ sentry_shutdown(void)
     sentry__mutex_unlock(&g_options_mutex);
 
     if (options) {
-        bool clean_shutdown = true;
+        size_t dumped_envelopes = 0;
         if (options->transport) {
             // TODO: make this configurable
-            clean_shutdown = sentry__transport_shutdown(
-                options->transport, SENTRY_DEFAULT_SHUTDOWN_TIMEOUT);
+            if (!sentry__transport_shutdown(
+                    options->transport, SENTRY_DEFAULT_SHUTDOWN_TIMEOUT)) {
+                SENTRY_WARN("transport did not shut down cleanly");
+            }
+            dumped_envelopes = sentry__transport_dump_queue(
+                options->transport, options->run);
```

Comment:
Could you even move this into the conditional block above?

Reply:
We are pretty much guaranteed to have 0 dumped envelopes on successful shutdown. But lets be defensive here anyway.